front/getimpacticon.php

```diff
@@ -0,0 +1,59 @@
+<?php
```

Comment:
You should use `front/pluginimage.send.php` instead of a custom file. It handles some security checks.

Reply:
I wasn't aware of `front/pluginimage.send.php`.

However, it does not seem to work for files in subfolders.

Testing the following URL:
`http://localhost/glpi/10.0-bugfixes/front/pluginimage.send.php?plugin=genericobject&name=impact_icons/PluginGenericobjectTestimpact_5e5cd4443b12c4.83229414test.png`
-> The `basename($_GET["name"]) != $_GET["name"]` test fails

`basename($_GET["name"])` -> `PluginGenericobjectTestimpact_5e5cd4443b12c4.83229414test.png`
`$_GET["name"]` -> `impact_icons/PluginGenericobjectTestimpact_5e5cd4443b12c4.83229414test.png`

Reply:
Indeed.
Solution is to, either, add `Document::isImage()` checks, at least on file reading, and probably also on file uploading, either store images not in a subdirectory, to be able to use the GLPI script.

This check was added for security reasons on GLPI, so I guess it is preferable to do it also here.

Reply:
Done.
It would be great if `front/pluginimage.send.php` could support subfolder, that would allow to delete this front file in a later version of the plugin.